### PR TITLE
feat(183320070): add babel and webpack presets to their own files

### PR DESCRIPTION
**Pull request by demogar**
# Feature

## Feature Description

The a goal is to have a way to test all the different ways to implement a `<BasicButton />`, with all the available permutations and variations.

- Modified the webpack config
- Modified the Jest config
- Modified the babel config
- Removed the original transformers (not needed)
- Created a way to have reports for our tests
- Created the specs for basic button

## Screenshot

```
 PASS   ProductPlan Atlas  tests/components/BasicButton.test.js
  BasicButton
    ✓ handles the click onClick handler (24 ms)
    ✓ renders the htmlForm property (7 ms)
    ✓ renders a different htmlType if provided (3 ms)
    when using different variants
      ✓ renders a default button (6 ms)
      ✓ renders a secondary button (4 ms)
      ✓ renders a ghost button (9 ms)
    when using icons
      when button with type default has a leading icon
        ✓ renders the icon (4 ms)
      when button with type default has a trailing icon
        ✓ renders the icon (2 ms)
      when button with type secondary has a leading icon
        ✓ renders the icon (2 ms)
      when button with type secondary has a trailing icon
        ✓ renders the icon (2 ms)
      when button with type ghost has a leading icon
        ✓ renders the icon (4 ms)
      when button with type ghost has a trailing icon
        ✓ renders the icon (2 ms)
```

Report:

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="ProductPlan Atlas - Unit Tests" tests="12" failures="0" errors="0" time="1.975">
  <testsuite name="BasicButton" errors="0" failures="0" skipped="0" timestamp="2022-10-12T17:44:55" time="1.4" tests="12">
    <testcase classname="BasicButton handles the click onClick handler" name="BasicButton handles the click onClick handler" time="0.024">
    </testcase>
    <testcase classname="BasicButton renders the htmlForm property" name="BasicButton renders the htmlForm property" time="0.007">
    </testcase>
    <testcase classname="BasicButton renders a different htmlType if provided" name="BasicButton renders a different htmlType if provided" time="0.003">
    </testcase>
    <testcase classname="BasicButton when using different variants renders a default button" name="BasicButton when using different variants renders a default button" time="0.006">
    </testcase>
    <testcase classname="BasicButton when using different variants renders a secondary button" name="BasicButton when using different variants renders a secondary button" time="0.004">
    </testcase>
    <testcase classname="BasicButton when using different variants renders a ghost button" name="BasicButton when using different variants renders a ghost button" time="0.009">
    </testcase>
    <testcase classname="BasicButton when using icons when button with type default has a leading icon renders the icon" name="BasicButton when using icons when button with type default has a leading icon renders the icon" time="0.004">
    </testcase>
    <testcase classname="BasicButton when using icons when button with type default has a trailing icon renders the icon" name="BasicButton when using icons when button with type default has a trailing icon renders the icon" time="0.002">
    </testcase>
    <testcase classname="BasicButton when using icons when button with type secondary has a leading icon renders the icon" name="BasicButton when using icons when button with type secondary has a leading icon renders the icon" time="0.002">
    </testcase>
    <testcase classname="BasicButton when using icons when button with type secondary has a trailing icon renders the icon" name="BasicButton when using icons when button with type secondary has a trailing icon renders the icon" time="0.002">
    </testcase>
    <testcase classname="BasicButton when using icons when button with type ghost has a leading icon renders the icon" name="BasicButton when using icons when button with type ghost has a leading icon renders the icon" time="0.004">
    </testcase>
    <testcase classname="BasicButton when using icons when button with type ghost has a trailing icon renders the icon" name="BasicButton when using icons when button with type ghost has a trailing icon renders the icon" time="0.002">
    </testcase>
  </testsuite>
</testsuites>
```

## Ticket

https://www.pivotaltracker.com/story/show/183320070

## Checklist

- [x] Tests
- [ ] Validated Against Design

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "gen:docs": "node scripts/generateComponentData.js",
     "gen:docs-watch": "npm run gen:docs -- --watch",
     "build:docs": "node scripts/build.js",
-    "test": "node scripts/test.js",
+    "test": "jest",
```
Comment (demogar) [inline package.json:19]: we can tweak this later, if necessary.

File: package.json [hunk 2]
```diff
@@ -42,101 +42,19 @@
       "last 1 safari version"
     ]
   },
-  "jest": {
```
Comment (demogar) [inline package.json:45]: moved all these configurations to their own files.

File: package.json [hunk 3]
```diff
@@ -6,8 +6,8 @@
   "main": "dist/components/index.js",
   "dependencies": {
     "prismjs": "^1.29.0",
-    "react": "16.12.0",
-    "react-dom": "16.12.0"
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
   },
```
Comment (demogar) [inline package.json:11]: ⚠️⚠️⚠️  `16.12.0` -> `18.2.0`

File: src/index.js [hunk 4]
```diff
@@ -1,10 +1,9 @@
...
 import './index.css';
 import Docs from './docs/Docs';
 
-ReactDOM.render(
-  <Docs />,
-  document.getElementById('root')
-);
+const container = document.getElementById('root');
+const root = createRoot(container); // createRoot(container!) if you use TypeScript
+root.render(<Docs />);
```
Comment (demogar) [inline src/index.js:9]: this was part of the React/React DOM upgrade.

**Review by kvuzet — APPROVED** [review]
ship it!

**Pull request merged**